### PR TITLE
add geolocation support #5

### PR DESCRIPTION
**Pull request by NaitYoussef**
Add geolocation support so we can return near restaurants #5

### CODE REVIEW

**Comment by zeroual** [comment]
@ynait do you need a code review

**Comment by NaitYoussef** [comment]
I'm good, thanks, I'll let you know when I'll need one ;)

File: frontend/src/app/home/home.component.ts [hunk 1]
```diff
@@ -11,12 +12,17 @@ export class HomeComponent implements OnInit {
...
+  constructor(private restaurantsService: RestaurantsService, private locationService: GeolocationService) {
   }
 
   ngOnInit() {
-    this.restaurantsService.getNearbyRestaurants()
-      .then(restaurants => this.restaurants = restaurants);
+
+      let promise = this.locationService.getCurrentPosition();
+      promise.then((crd) => {
+        alert("New coordinates youssef are: " + crd.latitude + " and " + crd.longitude);
```
Comment (zeroual) [inline frontend/src/app/home/home.component.ts]: `alert("New coordinates youssef are: " + crd.latitude + " and " + crd.longitude);`
hahhaha

**Review by zeroual — CHANGES_REQUESTED** [review]

File: frontend/src/app/home/home.component.ts (hunk 1 above)
Comment (zeroual) [reply frontend/src/app/home/home.component.ts]: Goood work @NaitYoussef  ❤️

File: frontend/src/app/shared/Coordinates.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface Coordinates {
+    latitude: number;
+    longitude: number;
+}
```
Comment (zeroual) [inline frontend/src/app/shared/Coordinates.ts:4]: Restaurant has already lat & lng ; maybe one day we can use this interface

File: frontend/src/app/home/home.component.ts (hunk 1 above)
Comment (NaitYoussef) [reply frontend/src/app/home/home.component.ts]: hahaha I'm going to remove it, thanks 😉

File: frontend/src/app/shared/Coordinates.ts (hunk 2 above)
Comment (NaitYoussef) [reply frontend/src/app/shared/Coordinates.ts:4]: Yes it's a good idea

**Review by berrada-zouhair — APPROVED** [review]
Nice PR @NaitYoussef

**Comment by zeroual** [comment]
@NaitYoussef  we need that you rebase in order to merge this PR

**Review by zeroual — APPROVED** [review]

**Comment by zeroual** [comment]
LGTM

**Pull request merged**